Core/GDCore/IDE/Project/ProjectResourcesCopier.cpp

```diff
@@ -25,28 +25,29 @@ bool ProjectResourcesCopier::CopyAllResourcesTo(
     bool updateOriginalProject,
     bool preserveAbsoluteFilenames,
     bool preserveDirectoryStructure) {
+
+  gd::Project& project = originalProject;
+  if (!updateOriginalProject) {
+    std::shared_ptr<gd::Project> clonedProject(new gd::Project(originalProject));
+    project = *clonedProject;
+  }
+
   // Check if there are some resources with absolute filenames
-  gd::ResourcesAbsolutePathChecker absolutePathChecker(originalProject.GetResourcesManager(), fs);
-  gd::ResourceExposer::ExposeWholeProjectResources(originalProject, absolutePathChecker);
+  gd::ResourcesAbsolutePathChecker absolutePathChecker(project.GetResourcesManager(), fs);
+  gd::ResourceExposer::ExposeWholeProjectResources(project, absolutePathChecker);
 
-  auto projectDirectory = fs.DirNameFrom(originalProject.GetProjectFile());
+  auto projectDirectory = fs.DirNameFrom(project.GetProjectFile());
   std::cout << "Copying all resources from " << projectDirectory << " to "
             << destinationDirectory << "..." << std::endl;
 
   // Get the resources to be copied
-  gd::ResourcesMergingHelper resourcesMergingHelper(originalProject.GetResourcesManager(), fs);
+  gd::ResourcesMergingHelper resourcesMergingHelper(project.GetResourcesManager(), fs);
   resourcesMergingHelper.SetBaseDirectory(projectDirectory);
   resourcesMergingHelper.PreserveDirectoriesStructure(
       preserveDirectoryStructure);
   resourcesMergingHelper.PreserveAbsoluteFilenames(
       preserveAbsoluteFilenames);
-
-  if (updateOriginalProject) {
-    gd::ResourceExposer::ExposeWholeProjectResources(originalProject, resourcesMergingHelper);
-  } else {
-    std::shared_ptr<gd::Project> project(new gd::Project(originalProject));

```

Comment:
Note that I think this is a overly complicated way of writing:
`gd::Project project = originalProject; //I.e.: you make a copy`